os-eap-probes/added/probes/probe/eap/dmr.py

```diff
@@ -146,3 +147,88 @@ def evaluate(self, results):
 
         return (min(status), messages)
 
+class HealthCheckProbe(DmrProbe):
+    """
+    Basic EAP probe which uses the DMR interface to query server state.  It
+    defines tests for server status, boot errors and deployment status.
+    """
+
+    def __init__(self):
+        super(HealthCheckProbe, self).__init__(
+            [
+                HealthCheckTest()
```

Comment:
If this uses dmr, would it be better to incorporate it into the existing dmr probe?  Also, do we need a similar test for the jolokia probes (I'm okay if we don't, as I think the jolokia probe is only used in places where the test fields aren't available through dmr, and dmr performs better)?

Reply:
oh, the reason i suggest this is because there's some batching that goes on, so we only send a single query to dmr.

Reply:
@rcernich Adding this test as another step in the existing batch was my initial approach. But I discovered that it is not feasible to use a batch with this op in it along with the other ops. The problem is if this op fails because the subsystem is not present, that will affect the results of the other ops in the batch. So even though this test would be able to deal with the failure of its own step (i.e. see if it's due to the extension not being present and if so ignore it), the other tests would fail.

I was even able to work around that problem easily enough for CD 14. But then I discovered that behavior was subtly different for previous EAP releases. I don't recall now if it was impossible to work around that in all case (I think it was) or perhaps it just would have been crazy complex and brittle.  Anyway, I ultimately settled on adding another probe.

Reply:
Re a test for the jolokia probes, it's here:

https://github.com/jboss-openshift/cct_module/pull/280/files#diff-381311125c9c364650be8092f3111047R149

The way probe.eap.jolokia.EapProbe works it was not necessary to separate out the jolokia HealthCheckTest into a separate probe.